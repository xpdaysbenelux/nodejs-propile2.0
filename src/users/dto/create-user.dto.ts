import {
    IsEmail,
    IsString,
    IsOptional,
    ArrayNotEmpty,
    IsUUID,
} from 'class-validator';

export class CreateUserRequest {
    @IsEmail()
    readonly email: string;

    @IsUUID('4', { each: true })
    @ArrayNotEmpty()
    readonly roleIds: string[];

    @IsString()
    @IsOptional()
    readonly firstName?: string;

    @IsString()
    @IsOptional()
    readonly lastName?: string;
}
