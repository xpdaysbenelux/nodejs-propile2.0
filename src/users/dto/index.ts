export * from './create-user.dto';
export * from './get-user.dto';
export * from './get-users.dto';
export * from './update-user.dto';
