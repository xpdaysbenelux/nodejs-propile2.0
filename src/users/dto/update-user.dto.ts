import { IsUUID, IsString, IsOptional, ArrayNotEmpty } from 'class-validator';
import { ApiProperty } from '@nestjs/swagger';

export class UpdateUserRequest {
    @IsUUID('4', { each: true })
    @IsOptional()
    @ArrayNotEmpty()
    readonly roleIds?: string[];

    @IsString()
    @IsOptional()
    readonly firstName?: string;

    @IsString()
    @IsOptional()
    readonly lastName?: string;
}

export class UserIdParam {
    @ApiProperty({ required: true })
    @IsUUID('4')
    readonly userId: string;
}
