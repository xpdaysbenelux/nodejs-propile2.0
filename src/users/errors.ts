import {
    BadRequestException,
    NotFoundException,
    MethodNotAllowedException,
} from '@nestjs/common';

export class EmailAlreadyInUse extends BadRequestException {
    constructor() {
        super('A user with this email already exists', 'EMAIL_ALREADY_IN_USE');
    }
}

export class RoleNotFound extends BadRequestException {
    constructor() {
        super('One of the roles passed was not found', 'ROLE_NOT_FOUND');
    }
}

export class UserNotFound extends NotFoundException {
    constructor() {
        super('The given user was not found', 'USER_NOT_FOUND');
    }
}

export class AccountAlreadyActive extends MethodNotAllowedException {
    constructor() {
        super(
            'The given user already has an active acount',
            'ACCOUNT_ALREADY_ACTIVE',
        );
    }
}
