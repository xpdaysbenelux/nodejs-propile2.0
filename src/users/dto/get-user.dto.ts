import { BaseEntityResponse } from '../../_shared/dto';
import { UserState } from '../../_shared/constants';

export class UserResponse extends BaseEntityResponse {
    readonly email: string;
    readonly state: UserState;
    readonly roles: UserResponseRole[];
    readonly firstName?: string;
    readonly lastName?: string;
}

class UserResponseRole {
    readonly id: string;
    readonly name: string;
}
