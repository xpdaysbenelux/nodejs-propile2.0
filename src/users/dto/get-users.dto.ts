import { IsOptional, IsEnum } from 'class-validator';

import { PagingQuery, PagingMeta } from '../../_shared/dto';
import { UserResponse } from './get-user.dto';

export enum UsersSortColumns {
    Email = 'email',
    State = 'state',
    CreatedAt = 'createdAt',
    UpdatedAt = 'updatedAt',
    CreatedBy = 'createdBy',
    UpdatedBy = 'updatedBy',
    FirstName = 'firstName',
    LastName = 'lastName',
}

export class GetUsersRequestQuery extends PagingQuery {
    @IsOptional()
    @IsEnum(UsersSortColumns)
    readonly sortBy?: UsersSortColumns;
}

export class GetUsersResponse {
    readonly meta: PagingMeta;
    readonly data: UserResponse[];
}
