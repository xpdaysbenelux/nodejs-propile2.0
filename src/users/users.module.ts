import { Module } from '@nestjs/common';
import { JwtModule } from '@nestjs/jwt';

import { UsersController } from './users.controller';
import { UsersService } from './users.service';
import { DatabaseModule } from '../database';
import { Config } from '../config';
import { UsersQueries } from './users.queries';
import { MailerModule } from '../mailer/mailer.module';

@Module({
    imports: [
        DatabaseModule,
        JwtModule.register({
            secret: Config.jwt.secret,
        }),
        MailerModule,
    ],
    controllers: [UsersController],
    providers: [UsersService, UsersQueries],
})
export class UsersModule {}
