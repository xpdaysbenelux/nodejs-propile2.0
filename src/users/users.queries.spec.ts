import { Test } from '@nestjs/testing';
import * as faker from 'faker';

import { UsersQueries } from './users.queries';
import { DatabaseModule, UserRepository } from '../database';
import { GetUsersRequestQuery, UsersSortColumns } from './dto';
import { SortDirection } from '../_shared/constants';

describe('UsersQueries', () => {
    let userRepository: UserRepository;
    let usersQueries: UsersQueries;

    beforeAll(async () => {
        const module = await Test.createTestingModule({
            imports: [DatabaseModule],
            providers: [UsersQueries],
        }).compile();

        usersQueries = module.get(UsersQueries);
        userRepository = module.get(UserRepository);
    });

    afterAll(() => {
        userRepository.manager.connection.close();
    });

    describe('getUser', () => {
        it('should return the requested user correctly', async () => {
            const result = await usersQueries.getUser(
                'c4cb4582-1e97-4e3e-9d49-c744c8c1a229',
            );
            expect(result).toMatchSnapshot();
        });

        it('should return nothing if the requested user does not exist', async () => {
            const result = await usersQueries.getUser(faker.random.uuid());
            expect(result).toMatchSnapshot();
        });
    });

    describe('getUsers', () => {
        const testQueries: GetUsersRequestQuery[] = [
            {},
            { skip: 2, take: 2 },
            { search: 'user1' },
            { search: 'DEVELOPMENT' },
            { search: 'Tom' },
            {
                sortBy: UsersSortColumns.CreatedAt,
                sortDirection: SortDirection.Ascending,
            },
        ];

        testQueries.forEach((query, index) => {
            it(`should return a paged list of users: #${index}`, async () => {
                const result = await usersQueries.getUsers(query);
                expect(result).toMatchSnapshot();
            });
        });

        it('should return an empty list if no users found', async () => {
            const result = await usersQueries.getUsers({
                search: 'nonsensicaljibberish',
            });
            expect(result).toMatchSnapshot();
        });
    });
});
