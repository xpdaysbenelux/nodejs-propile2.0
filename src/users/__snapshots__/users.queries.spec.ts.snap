// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersQueries getUser should return nothing if the requested user does not exist 1`] = `undefined`;

exports[`UsersQueries getUser should return the requested user correctly 1`] = `
User {
  "createdAt": 2020-01-01T08:21:59.730Z,
  "createdBy": null,
  "email": "development@icapps.com",
  "firstName": null,
  "id": "c4cb4582-1e97-4e3e-9d49-c744c8c1a229",
  "lastName": null,
  "roles": Array [
    Role {
      "id": "0c1510ab-5dca-41e2-8912-ee165140ae90",
      "name": "Admin",
    },
  ],
  "state": "ACTIVE",
  "updatedAt": 2020-01-01T08:21:59.730Z,
  "updatedBy": null,
}
`;

exports[`UsersQueries getUsers should return a paged list of users: #0 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2020-01-01T10:21:59.730Z,
      "createdBy": "development@icapps.com",
      "email": "user2@silvernext.com",
      "firstName": "Benno",
      "id": "45c2c554-b06a-4209-aba8-b097cb5f8566",
      "lastName": "Meysmans",
      "roles": Array [
        Role {
          "id": "d02a468f-0225-463d-a9a4-8c7d05cbe62c",
          "name": "Support",
        },
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "ACTIVE",
      "updatedAt": 2020-01-05T08:21:59.730Z,
      "updatedBy": "development@icapps.com",
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging1@silvernext.com",
      "firstName": null,
      "id": "951328c3-ab15-4b58-b759-4695a8eb59ca",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging2@silvernext.com",
      "firstName": null,
      "id": "1ca59e34-00ea-42df-b3c7-c36cabb448c2",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging4@silvernext.com",
      "firstName": null,
      "id": "5e2b0f76-caab-4d27-bc11-91eea1190fdb",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging5@silvernext.com",
      "firstName": null,
      "id": "18996066-1711-4c87-a2a0-e0c288ee1ff3",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging6@silvernext.com",
      "firstName": null,
      "id": "310f59f6-5456-4798-8ce0-48bc7cbafbb0",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging9@silvernext.com",
      "firstName": null,
      "id": "85140bf6-214e-40ee-a79f-dc8c4a4b5c3a",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging11@silvernext.com",
      "firstName": null,
      "id": "31e1634e-f92e-4fd2-ad58-f172b94b61d9",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging13@silvernext.com",
      "firstName": null,
      "id": "0b84e599-61db-4757-8c80-754a41be8451",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging14@silvernext.com",
      "firstName": null,
      "id": "10c93450-010c-471a-963d-14f67c84df41",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging15@silvernext.com",
      "firstName": null,
      "id": "4264fdef-899a-4861-aec3-0a8816465d07",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging16@silvernext.com",
      "firstName": null,
      "id": "63965bd7-ea40-47bd-9d5f-9c09ebce3a4f",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging17@silvernext.com",
      "firstName": null,
      "id": "043c6044-7302-48e8-ad37-bbd0b6e5ddd1",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging18@silvernext.com",
      "firstName": null,
      "id": "3cd2ef87-c542-48bd-b763-06ca71bd0b68",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging19@silvernext.com",
      "firstName": null,
      "id": "82ac0573-73bc-4da2-919a-fb8094da444c",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging20@silvernext.com",
      "firstName": null,
      "id": "45a16899-d259-4982-9040-3abd3d11b00e",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging21@silvernext.com",
      "firstName": null,
      "id": "417d2d73-41a0-43f2-9eef-800cab59209e",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging23@silvernext.com",
      "firstName": null,
      "id": "2df551b1-f385-4f72-862b-1870e41943c3",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging24@silvernext.com",
      "firstName": null,
      "id": "5df6c5e4-3d6e-40d5-9401-6d0db7340bc7",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging26@silvernext.com",
      "firstName": null,
      "id": "53558240-8094-4c3f-9028-e4626be150a9",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
  ],
  "meta": Object {
    "count": 20,
    "skip": 0,
    "totalCount": 32,
  },
}
`;

exports[`UsersQueries getUsers should return a paged list of users: #1 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging13@silvernext.com",
      "firstName": null,
      "id": "0b84e599-61db-4757-8c80-754a41be8451",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging14@silvernext.com",
      "firstName": null,
      "id": "10c93450-010c-471a-963d-14f67c84df41",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
  ],
  "meta": Object {
    "count": 2,
    "skip": 2,
    "totalCount": 32,
  },
}
`;

exports[`UsersQueries getUsers should return a paged list of users: #2 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2020-01-01T08:21:59.730Z,
      "createdBy": "development@icapps.com",
      "email": "user1@silvernext.com",
      "firstName": "Tom",
      "id": "49affd08-f461-4e80-899f-7612e971ff12",
      "lastName": "De Buyser",
      "roles": Array [
        Role {
          "id": "0c1510ab-5dca-41e2-8912-ee165140ae90",
          "name": "Admin",
        },
      ],
      "state": "ACTIVE",
      "updatedAt": 2020-01-01T08:21:59.730Z,
      "updatedBy": "development@icapps.com",
    },
  ],
  "meta": Object {
    "count": 1,
    "skip": 0,
    "totalCount": 1,
  },
}
`;

exports[`UsersQueries getUsers should return a paged list of users: #3 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2020-01-01T08:21:59.730Z,
      "createdBy": null,
      "email": "development@icapps.com",
      "firstName": null,
      "id": "c4cb4582-1e97-4e3e-9d49-c744c8c1a229",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "0c1510ab-5dca-41e2-8912-ee165140ae90",
          "name": "Admin",
        },
      ],
      "state": "ACTIVE",
      "updatedAt": 2020-01-01T08:21:59.730Z,
      "updatedBy": null,
    },
  ],
  "meta": Object {
    "count": 1,
    "skip": 0,
    "totalCount": 1,
  },
}
`;

exports[`UsersQueries getUsers should return a paged list of users: #4 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2020-01-01T08:21:59.730Z,
      "createdBy": "development@icapps.com",
      "email": "user1@silvernext.com",
      "firstName": "Tom",
      "id": "49affd08-f461-4e80-899f-7612e971ff12",
      "lastName": "De Buyser",
      "roles": Array [
        Role {
          "id": "0c1510ab-5dca-41e2-8912-ee165140ae90",
          "name": "Admin",
        },
      ],
      "state": "ACTIVE",
      "updatedAt": 2020-01-01T08:21:59.730Z,
      "updatedBy": "development@icapps.com",
    },
  ],
  "meta": Object {
    "count": 1,
    "skip": 0,
    "totalCount": 1,
  },
}
`;

exports[`UsersQueries getUsers should return a paged list of users: #5 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2019-01-01T08:32:59.730Z,
      "createdBy": "development@icapps.com",
      "email": "user4@silvernext.com",
      "firstName": null,
      "id": "c356fce8-9f95-4c69-9e71-8f4d74f49b8f",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "ACTIVE",
      "updatedAt": 2020-01-02T08:59:59.730Z,
      "updatedBy": "development@icapps.com",
    },
    User {
      "createdAt": 2020-01-01T08:21:59.730Z,
      "createdBy": "development@icapps.com",
      "email": "user1@silvernext.com",
      "firstName": "Tom",
      "id": "49affd08-f461-4e80-899f-7612e971ff12",
      "lastName": "De Buyser",
      "roles": Array [
        Role {
          "id": "0c1510ab-5dca-41e2-8912-ee165140ae90",
          "name": "Admin",
        },
      ],
      "state": "ACTIVE",
      "updatedAt": 2020-01-01T08:21:59.730Z,
      "updatedBy": "development@icapps.com",
    },
    User {
      "createdAt": 2020-01-01T08:21:59.730Z,
      "createdBy": "development@icapps.com",
      "email": "user3@silvernext.com",
      "firstName": "Jens",
      "id": "db5e788c-2642-4c40-a63a-f13675048ddc",
      "lastName": "De Craecker",
      "roles": Array [
        Role {
          "id": "d02a468f-0225-463d-a9a4-8c7d05cbe62c",
          "name": "Support",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-01T08:21:59.730Z,
      "updatedBy": "development@icapps.com",
    },
    User {
      "createdAt": 2020-01-01T08:21:59.730Z,
      "createdBy": null,
      "email": "development@icapps.com",
      "firstName": null,
      "id": "c4cb4582-1e97-4e3e-9d49-c744c8c1a229",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "0c1510ab-5dca-41e2-8912-ee165140ae90",
          "name": "Admin",
        },
      ],
      "state": "ACTIVE",
      "updatedAt": 2020-01-01T08:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-01T10:21:59.730Z,
      "createdBy": "development@icapps.com",
      "email": "user2@silvernext.com",
      "firstName": "Benno",
      "id": "45c2c554-b06a-4209-aba8-b097cb5f8566",
      "lastName": "Meysmans",
      "roles": Array [
        Role {
          "id": "d02a468f-0225-463d-a9a4-8c7d05cbe62c",
          "name": "Support",
        },
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "ACTIVE",
      "updatedAt": 2020-01-05T08:21:59.730Z,
      "updatedBy": "development@icapps.com",
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging13@silvernext.com",
      "firstName": null,
      "id": "0b84e599-61db-4757-8c80-754a41be8451",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging14@silvernext.com",
      "firstName": null,
      "id": "10c93450-010c-471a-963d-14f67c84df41",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging15@silvernext.com",
      "firstName": null,
      "id": "4264fdef-899a-4861-aec3-0a8816465d07",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging17@silvernext.com",
      "firstName": null,
      "id": "043c6044-7302-48e8-ad37-bbd0b6e5ddd1",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging18@silvernext.com",
      "firstName": null,
      "id": "3cd2ef87-c542-48bd-b763-06ca71bd0b68",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging20@silvernext.com",
      "firstName": null,
      "id": "45a16899-d259-4982-9040-3abd3d11b00e",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging21@silvernext.com",
      "firstName": null,
      "id": "417d2d73-41a0-43f2-9eef-800cab59209e",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging23@silvernext.com",
      "firstName": null,
      "id": "2df551b1-f385-4f72-862b-1870e41943c3",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging24@silvernext.com",
      "firstName": null,
      "id": "5df6c5e4-3d6e-40d5-9401-6d0db7340bc7",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging26@silvernext.com",
      "firstName": null,
      "id": "53558240-8094-4c3f-9028-e4626be150a9",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging2@silvernext.com",
      "firstName": null,
      "id": "1ca59e34-00ea-42df-b3c7-c36cabb448c2",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging4@silvernext.com",
      "firstName": null,
      "id": "5e2b0f76-caab-4d27-bc11-91eea1190fdb",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging5@silvernext.com",
      "firstName": null,
      "id": "18996066-1711-4c87-a2a0-e0c288ee1ff3",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging6@silvernext.com",
      "firstName": null,
      "id": "310f59f6-5456-4798-8ce0-48bc7cbafbb0",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-01-02T09:21:59.730Z,
      "createdBy": null,
      "email": "paging11@silvernext.com",
      "firstName": null,
      "id": "31e1634e-f92e-4fd2-ad58-f172b94b61d9",
      "lastName": null,
      "roles": Array [
        Role {
          "id": "81388934-7613-4f66-a225-a0d246f65e61",
          "name": "Medewerker",
        },
      ],
      "state": "REGISTERING",
      "updatedAt": 2020-01-02T09:21:59.730Z,
      "updatedBy": null,
    },
  ],
  "meta": Object {
    "count": 20,
    "skip": 0,
    "totalCount": 32,
  },
}
`;

exports[`UsersQueries getUsers should return an empty list if no users found 1`] = `
Object {
  "data": Array [],
  "meta": Object {
    "count": 0,
    "skip": 0,
    "totalCount": 0,
  },
}
`;
